libs/muddle/src/peer_selector.cpp

```diff
@@ -233,10 +234,12 @@ void PeerSelector::ResolveAddresses(Addresses const &addresses)
     // make the call to the remote service
     auto promise = rpc_client_.CallSpecificAddress(address, RPC_MUDDLE_DISCOVERY,
                                                    DiscoveryService::CONNECTION_INFORMATION);
+    // lookup the peer information
+    auto const &peer_data = peers_info_[address];
 
     // wrap the promise is a task
     auto task = std::make_shared<PromiseTask>(
-        std::move(promise),
+        std::move(promise), CalculatePromiseTimeout(peer_data.consecutive_failures),
```

Comment:
Nitpicking: doesn't this overload accept an lvalue promise?

Reply:
No that is a good spot